hewr/R/process_state_forecast.R

```diff
@@ -150,30 +108,57 @@ to_tidy_draws_timeseries <- function(tidy_forecast,
   dplyr::bind_rows(
     transformed_obs,
     tidy_forecast
-  )
+  ) |>
+    dplyr::select(.draw, everything())
 }
 
+join_and_calc_prop <- function(model_1, model_2) {
+  inner_join(
+    pivot_wider(model_1, names_from = ".variable", values_from = ".value"),
+    pivot_wider(model_2, names_from = ".variable", values_from = ".value"),
+    by = join_by(.draw, date, geo_value, disease)
+  ) |>
+    mutate(prop_disease_ed_visits = observed_ed_visits /
+      (observed_ed_visits + other_ed_visits)) |>
+    pivot_longer(-c(starts_with("."), date, geo_value, disease),
+      names_to = ".variable", values_to = ".value"
+    ) |>
+    drop_na()
```

Comment:
```suggestion
  dplyr::inner_join(
    tidyr::pivot_wider(model_1, names_from = ".variable", values_from = ".value"),
    tidyr::pivot_wider(model_2, names_from = ".variable", values_from = ".value"),
    by = dplyr::join_by(.data$.draw, .data$date, .data$geo_value, .data$disease)
  ) |>
    mutate(prop_disease_ed_visits = .data$observed_ed_visits /
      (.data$observed_ed_visits + .data$other_ed_visits)) |>
    tidyr::pivot_longer(-c(starts_with("."), date, geo_value, disease),
      names_to = ".variable", values_to = ".value"
    ) |>
    tidyr::drop_na()
```

Feels like `with_prop_disease_ed_visits()` could've been used here and a couple other places. Any strong reasons you eliminated it? Local readability of column names?

Reply:
It only gets used a couple of places, and the steps before and after the calculation are different in each case, so I didn't see much utility in writing the one-line function.